packages/fxa-auth-server/lib/types.ts

```diff
@@ -71,8 +60,22 @@ export interface ProfileClient {
   updateDisplayName(uid: string, name: string): Promise<void>;
 }
 
+export interface AuthLogger extends Logger {
+  (tags: string | string[], data?: string | object): void;
+
+  begin(location: string, request: AuthRequest): void;
+
+  notifyAttachedServices(
+    serviceName: string,
+    request: AuthRequest,
+    data: Record<string, any>
+  ): Promise<void>;
+}
+
 // Container token types
+// eslint-disable-next-line @typescript-eslint/no-redeclare
 export const AuthLogger = new Token<AuthLogger>('AUTH_LOGGER');
 export const AuthFirestore = new Token<Firestore>('AUTH_FIRESTORE');
 export const AppConfig = new Token<ConfigType>('APP_CONFIG');
+// eslint-disable-next-line @typescript-eslint/no-redeclare
 export const ProfileClient = new Token<ProfileClient>('PROFILE_CLIENT');
```

Comment:
Hm... I guess one `AuthLogger` (and `ProfileClient`) is an `interface` which is a TS-only thing that gets removed when TS is compiled to JS, so in the JS, we will only have one variable here. Are you avoiding renaming any of them, since both the interface and types are exported and used elsewhere, requiring a large number of variable name changes throughout the source code?

Reply:
Yes - avoiding renaming. I went down the path of attempting a rename and I decided to leave that as out of scope. Our typenames should generally not be the same as variable names in the future. IProfileClient/IAuthLogger is always a good way to handle this.